scene/3d/navigation_region_3d.cpp

```diff
@@ -171,7 +171,7 @@ void NavigationRegion3D::bake_navigation_mesh(bool p_on_thread) {
 	BakeThreadsArgs *args = memnew(BakeThreadsArgs);
 	args->nav_region = this;
 
-	if (p_on_thread) {
+	if (p_on_thread && OS::get_singleton()->can_use_threads()) {
```

Comment:
With the above approach, it is possible the user will ask navigation mesh to bake on the thread but Godot will silently do it off thread. IMHO that's wrong. As a user, in the situation when I `bake(true)` and the platform does not support threading, I'd expect a warning (in case of fallback to off-thread baking) or error (in case we block such scenario entirely).

I guess lots of users used to not care about warnings, so a warning about auto fallback may be overlooked - yet it may have a severe impact on performance. Therefore I'd say we should issue an error asking one to not use `on_thread=true`.

Reply:
I guess a user warn_print() in the editor cannot hurt while a red error might be too much. Should still turn threaded baking off automatically in my opinion cause on_thread is true by default. A crash or not baked navmesh is worse than a framerate stutter. If framerate has problems on certain export platforms users usually begin to look at warnings and logs to notice this problem.

Reply:
Ok, let's add warning - if someone will complain we will make it an error then.